src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java

```diff
@@ -148,6 +183,7 @@ public Consumer<Message<String>> consumeRun() {
                 this.handleNonCancellationException(resultContext, e, rootReporter);
                 throw new ComputationException(String.format("%s: %s", NotificationService.getFailedMessage(getComputationType()), e.getMessage()), e.getCause());
             } finally {
+                processDebug(resultContext);
```

Comment:
Shouldn't you test this here instead of in the `processDebug` function ?
```suggestion
            if (Boolean.TRUE.equals(resultContext.getRunContext().getDebug())) {
                processDebug(resultContext);
            }
```

Reply:
YES